components/autofill/src/sync/address/incoming.rs

```diff
@@ -118,18 +126,18 @@ fn save_incoming_tombstones(
     signal: &dyn Interruptee,
 ) -> Result<()> {
     log::info!("staging {} incoming tombstones", incoming_tombstones.len());
-    let chunk_size = 1; // XXX - chunk_size of 1 seems wrong?
     sql_support::each_sized_chunk(
         &incoming_tombstones,
-        sql_support::default_max_variable_number() / chunk_size,
+        sql_support::default_max_variable_number(),
         |chunk, _| -> Result<()> {
+            let chunk_size = 1;
             let sql = format!(
                 "INSERT OR REPLACE INTO temp.addresses_tombstone_sync_staging (
                     guid
                 ) VALUES {}",
                 sql_support::repeat_multi_values(chunk.len(), chunk_size)
             );
-            let mut params = Vec::with_capacity(chunk.len() * chunk_size);
+            let mut params = Vec::with_capacity(chunk.len());
             for record in chunk {
                 signal.err_if_interrupted()?;
```

Comment:
While this isn't part of your patch, `err_if_interrupted` should move out of this loop, and just called once per chunk.

(and once you do that, this entire loop could be replaced with `let params: Vec<&dyn ToSql> = chunk.iter().map(|r| &r.guid as &dyn ToSql).collect();` - but I'm fine with keeping the loop explicit if you think it's clearer.